### DISCUSSION TITLE
Linear Regression

### DISCUSSION THREAD

**Discussion by maw-cu** [LCA-ActivityBrowser/activity-browser#808] [Ideas]
I have managed to implement linear regression in the activity browser, with the calculations done by scikit-learn, if anyone is interested.

**Comment by marc-vdm** [comment]
This sounds interesting! Could you explain a bit more about what you did, what you can do with the regression tool and how it works (both technically and for users)?

**Comment by maw-cu** [comment]
Well, sensitivity analysis methods can be classified as a) regression-based, b) variance-based, c) derivative-based, etc.....

It was while I was upgrading the sensitivity analysis section to include more methods that I thought of introducing linear regression to the activity browser. I thought that this would be a useful separate LCA result subtab (next to sensitivity analysis) for assessing less complex models with limited parameter interactions.

For now, I can express the model output, which can be CO2 emissions, as a function of input parameters through a linear relation. The final goal is to compute standardized regression coefficients (SRCs), which can function as delta or s-index coefficients to also rank input parameters according to their importance and influence on model output.
![image](https://user-images.githubusercontent.com/34771173/177956933-83afde8b-30c8-4f41-9a77-a86d39087275.png)

**Reply by marc-vdm** [reply]
Exciting stuff, though I must admit it's not entirely clear to me what the regression coefficients would be? In case of using ecoinvent, would you have ~20k regression variables? Could you perhaps refer to papers going further into the method?

**Comment by maw-cu** [comment]
Since I have someone's attention now. All these endeavors were due to my frustration with the Montecarlo simulation because it will give different output curves depending on how many reference flows you are assessing at a time (Issue #777 ).

Can you please address this issue, I would like to finish my PhD. sometime soon?

**Comment by bsteubing** [comment]
@maw-cu This is a great initiative! Would you be willing to explain what you did in a personal meeting? Please contact us on our Leiden email accounts for that. Then we could also discuss the possibility of making your features available within the AB.
